### DISCUSSION TITLE
[BUG?] Antd useSelect hook with mode['multiple']; issue -> unique "key" prop & UI

### DISCUSSION THREAD

**Discussion by w20k** [refinedev/refine#4958] [Q&A] [answered]
Hey Refine Devs/Community,

Node v.: v18.16.0
refine-core: @refinedev/core": "^4.38.4",
refine-antd: @refinedev/antd": "^5.34.2"
antd: "^5.9.2"

It feel like I'm missing something or that might be a bug with in 'multiple' mode, still digging, but if anyone had it and found how to fix it - would be grateful if you could share you hack ;)

I will try few things more and re-reding the docs as well, might be helpful 😄 Or I will hack it for now, but using build in tools seems more cleaner!

Code is simple, by the book:
```tsx
  const workerData = queryResult?.data?.data;
  const { selectProps: workTypesSelectProps } = useSelect({
    resource: 'work-types',
    optionLabel: 'name',
    defaultValue: workerData?.work_types.map((workType) => workType.id),
  });

........

<Select {...workTypesSelectProps} mode='multiple' size={'large'}  />

```

UI:
![Screenshot 2023-09-19 at 17 19 16](https://github.com/refinedev/refine/assets/1053320/71830a62-6efc-44a7-ad9f-7820e49d6780)

Error:
![Screenshot 2023-09-19 at 17 23 08](https://github.com/refinedev/refine/assets/1053320/93bee6bd-5550-47f3-9670-49ef363d2d5e)

**Comment by alicanerdurmaz** [comment]
Hello @w20k,
it's a long shot but can you try this, please? 

```ts
const workerData = queryResult?.data?.data || [];
```

However, to debug this issue, I need `workerData` and `work-types` values. Can you share it with me?

By the way, if you provide a producible example, I will be happy to debug.

**Reply by w20k** [reply]
Hey @alicanerdurmaz,

We're using Strapi (and strapi dataprovider) as a backend server (might be able to setup something similar with render - it will take time 😄 ) for a quick MVP. So, it's pretty much simple worker -> work_type model with one to many relation, where you could have a multiple work types for a single worker, like in example below.

It didn't help, the values are there, but for some reason there're not selected as a [multiple values].
<img width="580" alt="Screenshot 2023-09-19 at 18 30 33" src="https://github.com/refinedev/refine/assets/1053320/fab088d4-b860-41c0-a1e2-a7b87614940d">


```json
{
    "data": {
        "id": 1,
        "attributes": {
            "email": "work-test@test-work.com",
            "phone": "32432423",
            "is_active": true,
            "has_login": false,
            "description": "fsfsdf ",
            "consultant_id": "RHE-13123213",
            "createdAt": "2023-09-18T16:04:15.421Z",
            "updatedAt": "2023-09-18T16:36:11.648Z",
            "first_name": "Kiron ",
            "last_name": "Hrolino",
            "work_types": {
                "data": [
                    {
                        "id": 1,
                        "attributes": {
                            "reference": "siembra",
                            "name": "Siembra",
                            "createdAt": "2023-09-18T16:12:24.371Z",
                            "updatedAt": "2023-09-18T16:12:24.371Z"
                        }
                    },
                    {
                        "id": 2,
                        "attributes": {
                            "reference": "Reference",
                            "name": "Name",
                            "createdAt": "2023-09-18T16:16:04.493Z",
                            "updatedAt": "2023-09-18T16:16:04.493Z"
                        }
                    }
                ]
            }
        }
    },
    "meta": {}
}
```

work-types
```json
{
    "data": [
        {
            "id": 1,
            "attributes": {
                "reference": "siembra",
                "name": "Siembra",
                "createdAt": "2023-09-18T16:12:24.371Z",
                "updatedAt": "2023-09-18T16:12:24.371Z"
            }
        },
        {
            "id": 2,
            "attributes": {
                "reference": "Reference",
                "name": "Name",
                "createdAt": "2023-09-18T16:16:04.493Z",
                "updatedAt": "2023-09-18T16:16:04.493Z"
            }
        }
    ],
    "meta": {
        "pagination": {
            "page": 1,
            "pageSize": 2,
            "pageCount": 1,
            "total": 2
        }
    }
}
```

**Reply by w20k** [reply, marked as answer]
Found the option, how it's generated vie inferencer (aweomse thingy 😄 ), issue was related to antd form.item:
```tsx
 <Form.Item
          label={translate('workers.fields.work_types')}
          name={'work_types'}
          rules={[
            {
              required: true,
            },
          ]}
          getValueProps={(value: any[]) => {
            return {
              value: value?.map((item) => item?.id),
            };
          }}
          getValueFromEvent={(selected: string[]) => {
            return selected?.map((item) => ({ id: item }));
          }}
        >
          <Select mode='multiple' {...workTypesSelectProps} />
        </Form.Item>
```
<img width="338" alt="Screenshot 2023-09-20 at 00 18 51" src="https://github.com/refinedev/refine/assets/1053320/df689683-2ace-4ff1-8b40-7b027eb7c599">
